src/provisioning/api.md

```diff
@@ -0,0 +1,196 @@
+the appservice or webhook listeners.
+
+Requests should always be authenticated with the secret given in the config, inside the `Authorization` header.
+Requests being made on behalf of users (most provisioning APIs) should include the userId as a query parameter.
+
+```
+GET /v1/health?userId=%40Half-Shot%3Ahalf-shot.uk
+Authorization: Bearer secret
+```
+
+APIs are versioned independently so two endpoints on the latest version may not always have the same version.
+
+# APIs
+
+## GET /v1/health
+
+Request the status of the provisoning API.
+
+### Response
+
+```
+HTTP 200
+{}
+```
+
+Any other response should be considered a failed request (e.g. 404, 502 etc).
+
+## GET /v1/connectiontypes
+
+Request the connection types enabled for this bridge.
+
+### Response
+
+```json5
+{
+    "JiraProject": {
+        "type": "JiraProject", // The name of the connection
+        "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type for the connection
+        "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+        "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    }
+}
+```
+
+## GET /v1/{roomId}/connections
+
+Request the connections for a given room. The `{roomId}` parameter is the target Matrix room.
+
+### Response
+
+```json5
+[{
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}]
+```
+
+
+## GET /v1/{roomId}/connections/{id}
+
+Request details of a single connection. The `{roomId}` parameter is the target Matrix room.
+
+### Response
+
+```json5
+{
+
+    "type": "JiraProject", // The name of the connection
+    "eventType": "uk.half-shot.matrix-hookshot.jira.project", // Corresponds to the state type in the connection
+    "id": "opaque-unique-id", // An opaque ID used to refer to this connection. Should **NOT** be assumed to be stable.
+    "service": "jira", // or github, webhook. A human-readable service name to make things look pretty
+    "botUserId": "@hookshot:yourdomain.com", // The bot mxid for the service. Currently this is the sender_localpart, but may change in the future.
+    "config": {
+        // ... connection specific details, can be configured.
+    }
+}
+```
+
+## PUT /v1/{roomId}/connections/{type}
```

Comment:
is it possible to get an endpoint to list all capable jira projects for the authed user? Ideally in a format that can be picked apart and fed back to this PUT endpoint.

Likewise for github orgs/repos.

Reply:
Done in `32ac765` for JIRA, github coming up!